src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -284,7 +267,19 @@ public Configuration getConfiguration() {
 
     @Override
     public boolean isInMode(ShipMode mode) {
-        return mode.getConfiguration() == getConfiguration() && mode.getFormation() == getFormation();
+        return mode.getConfiguration() == getConfiguration() && (mode.getFormation() == getFormation()
+                || (mode.getFormation() == null && mode instanceof MutableShipMode
+                    && ((MutableShipMode) mode).toSelectChar == selectedChar));
```

Comment:
Either MutableShipMode, or Config.ShipMode, both of those have a char to check against